_drafts/_2023-04-14-memfault-matter.md

```diff
@@ -0,0 +1,358 @@
+  * Change `CONFIG_MEMFAULT_NCS_PROJECT_KEY` with a key you can get from [Memfault Cloud | Settings | General | Project Key](https://mflt.io/project-key).
+
+### Memfault UDP relay
+
+Minor modifications of <https://github.com/memfault/memfault-nRF9160-relay/tree/main/server> were required for IPv6 and the modified file is located in `server/` of <https://github.com/markus-becker-tridonic-com/matter-thread-memfault>.
+
+* The script has been changed to use an IPv6 socket.
+* The script joins the IPv6 multicast group on the relevant interface (here `utun3`, adapt to your interface name).
+
+```python
+INTERFACE = 'utun3'
+LOCAL_PORT = 20001
+BUFFER_SIZE = 508 + 28
+
+MEMFAULT_IPV6_GROUP_6 = 'ff05::f417'
+
+addrinfo = socket.getaddrinfo(MEMFAULT_IPV6_GROUP_6, None)[0]
+
+print(addrinfo)
+
+UDPServerSocket = socket.socket(addrinfo[0], type=socket.SOCK_DGRAM)
+UDPServerSocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+UDPServerSocket.bind(('', LOCAL_PORT))
+
+group_bin = socket.inet_pton(addrinfo[0], addrinfo[4][0])
+mreq = group_bin + struct.pack('@I', socket.if_nametoindex(INTERFACE))
+UDPServerSocket.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_JOIN_GROUP, mreq)
+
+print(f"Listening on interface {INTERFACE} port {LOCAL_PORT}")
+```
+
+### Build the light_bulb sample
+
+You can build the light_bulb sample according to [Nordic's documentation](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/2.3.0/nrf/samples/matter/light_bulb/README.html).
+
+## Run Border Router and Commission matter device
+
+Set up a border router and commission the matter light_bulb according to the [Nordic matter documentation](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/matter/index.html).
+
+## Run the UDP relay
+
+Execute the relay:
+
+```shell
+cd server
+python3 __main__.py
+```
+
+## Trigger a fault with the mflt CLI on the Nordic DK
+
+Connect to the UART shell on the DK with e.g. putty and issue e.g. `mflt test hardfault` or other faults from <https://docs.memfault.com/docs/mcu/zephyr-guide#testing-the-integration>.
+
+```shell
+> mflt test hardfault
+```
+
+Observe the transmission of the UDP packets to the UDP relay. Observe the UDP relay forwarding the information to the memfault cloud. And then see the chunks arriving in the [Memfault cloud | Chunks Debug](https://mflt.io/chunks-debug).
+
+<!--
+## Future topics
+
+* [x] MEMFAULT_NCS_DEVICE_ID_RUNTIME / memfault_ncs_device_id_set()
+  * [x] check functionality
+  * [ ] increase/limit size of the device ID
+* [ ] include dedicated matter/Thread metrics
+* [ ] track matter commissioning problems via memfault via BLE
+* [ ] derive certain memfault configs from matter/mcuboot configs or factory data, e.g.
+  * [ ] MEMFAULT_NCS_HW_VERSION
+  * [ ] MEMFAULT_NCS_FW_TYPE
+  * [ ] MEMFAULT_FW_VERSION / MEMFAULT_NCS_FW_VERSION_STATIC / MEMFAULT_NCS_FW_VERSION_AUTO
```

Comment:
I'm not sure I follow what this section is? Is this an outline of things people do after integrating, or topics you want to cover in future articles?